### PR TITLE
Wrap mipmap, scaleMode, wrapMode in getter/setter

### LINKED ISSUE #7550: BaseTexture wrapMode, scaleMode, mipmap should be getters/setters that increment dirtyStyleId
## Context

I was using `Graphics` with a texture fill and was surprised when the texture did not tile. I tried reproducing this with a lone `Graphics` but would not (unless you used very large coordinates - which was unrelated).

Screen Shot 2021-06-22 at 6 37 14 PM

Then I tried not rendering the existing tiles as sprites - and the problem went away. It turns out that `GraphicsGeometry` sets the `wrapMode` on the texture here:

https://github.com/pixijs/pixijs/blob/410a49c4f4e938d75a52a5129e98eb771780dc61/packages/graphics/src/GraphicsGeometry.ts#L509

However, if the texture has already been rendered once - the texture style would not be dirty. Since setting wrap-mode doesn't update the `dirtyStyleId`, the texture's wrap-mode wouldn't be updated in WebGL.

### Possible Solution

This can be fixed by

* making `wrapMode` a getter/setter that updates `dirtyStyleId` when you change it
* or calling `texture.baseTexture.update()` in `GraphicsGeometry`

I like the first option.

### Steps to Reproduce

In the following fiddle, I render the stage once with a single sprite (left). Then I add texture-tiling graphics (right) and re-render the scene.

https://jsfiddle.net/ShukantPal/dag0bhq9/

Screen Shot 2021-06-23 at 3 11 30 PM

The "Update texture" button will call `baseTexture.update()` and re-render the scene - and that fixes the problem:

Screen Shot 2021-06-23 at 3 11 52 PM

### Environment

- **`pixi.js` version**: 6.0.4
- **Running Example**: [https://jsfiddle.net/ShukantPal/dag0bhq9](https://jsfiddle.net/ShukantPal/dag0bhq9)

### PR DESCRIPTION
**Pull request by ShukantPal**
Fixes #7550 

Also: move documentation comments to properties

---
I'm back 👋

### CODE REVIEW

**Review by bigtimebuddy — COMMENTED** [review]
For the setters, probably be worth checking for changes so dirty is only if there's a meaningful change?

**Comment by ShukantPal** [comment]
Indeed, my bad - will add dirty checks.

File: packages/core/src/textures/BaseTexture.ts [hunk 1]
```diff
@@ -381,6 +351,53 @@ export class BaseTexture<R extends Resource = Resource, RO = IAutoDetectOptions>
...
+     * @default PIXI.settings.MIPMAP_TEXTURES
+     */
+    get mipmap(): MIPMAP_MODES
+    {
+        return this._mipmap;
+    }
+    set mipmap(value: MIPMAP_MODES)
+    {
+        this._mipmap = value;
+        this.dirtyStyleId++;
```
Comment (bigtimebuddy) [inline packages/core/src/textures/BaseTexture.ts:367]:
```suggestion
        if (this._mapmap !== value)
        {
            this._mipmap = value;
            this.dirtyStyleId++;
        }
```

File: packages/core/src/textures/BaseTexture.ts [hunk 2]
```diff
@@ -381,6 +351,53 @@ export class BaseTexture<R extends Resource = Resource, RO = IAutoDetectOptions>
...
+     * @default PIXI.settings.SCALE_MODE
+     */
+    get scaleMode(): SCALE_MODES
+    {
+        return this._scaleMode;
+    }
+    set scaleMode(value: SCALE_MODES)
+    {
+        this._scaleMode = value;
+        this.dirtyStyleId++;
```
Comment (bigtimebuddy) [inline packages/core/src/textures/BaseTexture.ts:383]:
```suggestion
        if (this._scaleMode !== value)
        {
            this._scaleMode = value;
            this.dirtyStyleId++;
        }
```

File: packages/core/src/textures/BaseTexture.ts [hunk 3]
```diff
@@ -381,6 +351,53 @@ export class BaseTexture<R extends Resource = Resource, RO = IAutoDetectOptions>
...
+     * @default PIXI.settings.WRAP_MODE
+     */
+    get wrapMode(): WRAP_MODES
+    {
+        return this._wrapMode;
+    }
+    set wrapMode(value: WRAP_MODES)
+    {
+        this._wrapMode = value;
+        this.dirtyStyleId++;
```
Comment (bigtimebuddy) [inline packages/core/src/textures/BaseTexture.ts:398]:
```suggestion
        if (this._wrapMode !== value)
        {
            this._wrapMode = value;
            this.dirtyStyleId++;
        }
```

**Review by GoodBoyDigital — APPROVED** [review]
👍  nice

**Review by bigtimebuddy — APPROVED** [review]

**Pull request merged**